js/main.js

```diff
@@ -1,183 +1,50 @@
-  DETAILS_DIALOG_EL.getElementsByClassName(
-    "dialog-title"
-  )[0].insertAdjacentHTML(
-    "beforeend",
-    `
-    <h2 class="card-name">${data["Card Display Name"]}</h2>
-    <p>Level: ${data["Card Level"]} | HP: ${data["Card HP"]} |  Element: ${data["Card Element"]} |  Card #: ${data["Collector Number"]}</p>
-    <p>Artist: ${data["Artist Credit"]} | Writer: ${data["Writer Credit"]}</p>
-  `
-  );
-  // Card
-  DETAILS_DIALOG_EL.getElementsByClassName("details-dialog-card")[0].src =
-    cardUrl;
-  // Clear + set card metadata
-  DETAILS_DIALOG_EL.getElementsByClassName("details-dialog-text")[0].innerHTML =
-    "";
-  DETAILS_DIALOG_EL.getElementsByClassName(
-    "details-dialog-text"
-  )[0].insertAdjacentHTML(
-    "beforeend",
-    `
-    <h3>${data["Attack 1 Name"]}</h3>
-    <p>${data["Attack 1 Description"]}</p>
-    <p><i>${data["Attack 1 Extended Lore"]}</i></p>
-    <h3>${data["Attack 2 Name"]}</h3>
-    <p>${data["Attack 2 Description"]}</p>
-    <p><i>${data["Attack 2 Extended Lore"]}</i></p>
-  `
-  );
-}
-
-function getCSVAndMaybeRenderCollection() {
+function getCSVData(callback = undefined) {
   Papa.parse(CSV_FILENAME, {
     download: true,
     //To treat the first row as column titles
     header: true,
     complete: (result) => {
       cards_data = result.data;
-      if (CURRENT_PAGE == "/collection.html") {
-        renderCards(cards_data, COLLECTIONS_MAIN_CONTENT);
+      if (callback) {
+        callback(cards_data);
       }
     },
   });
 }
 
-//Pulls n cards from the passed "cards" array using the Fisher-Yates Shuffle.
-function pullCards(cards, n) {
-  if (n <= 0) {
-    throw "n must be a positive integer bigger than 0";
-  }
-  let len = cards.length;
-  if (n > len) {
-    throw "Not enough elements in array";
-  }
-  let samples = [...cards];
-
-  while (len) {
-    let i = Math.floor(Math.random() * len--);
-    let j = samples[len];
-    samples[len] = samples[i];
-    samples[i] = j;
-  }
-  return samples.slice(0, n);
-}
-
-//Pulls "n" number of cards from the cards_data array and renders them in CARD_LIST.
-function pullAndRenderCards(n) {
-  let pulled = pullCards(cards_data, n);
-  renderCards(pulled, COLLECTIONS_MAIN_CONTENT, true);
-}
-
-//Renders a list of cards in the element specified in htmlLocation.
-//If replace is true, overwrites all elements inside htmlLocation.
-//else, adds the cards to the rest of the inner content.
-function renderCards(cards, htmlLocation, replace = false) {
-  if (replace) {
-    htmlLocation.innerHTML = "";
-  }
-  for (let i = 0; i < cards.length; i++) {
-    htmlLocation.insertAdjacentHTML(
-      "beforeend",
-      `<tcg-card card-id="${cards[i]["Collector Number"]}"></tcg-card>`
-    );
-  }
-}
-
 async function main() {
   await defineCardComponent();
-  getCSVAndMaybeRenderCollection();
+  getCSVData((cards_data) => {
+    if (CURRENT_PAGE == "/collection.html") {
+      renderCards(cards_data, COLLECTIONS_MAIN_CONTENT);
+    }
```

Comment:
I like this callback change - consolidates the current page checking in one spot